jscomp/runtime/caml_splice_call.ml

```diff
@@ -38,6 +38,21 @@ let spliceApply : obj -> obj -> obj = [%raw{|function(fn,args){
   return fn.apply(null,applied)
 }|}] 
 
+let spliceNewApply : obj -> obj -> obj = [%raw{|function (ctor,args){
+  var i, argLen;
+  argLen = args.length
+  var applied = []
```

Comment:
Oh, I'm sorry, but last time I didn't draw your attention to one more change besides the spread operator.

Then we use the hack with `Function.prototype.bind.apply(fn, args)` then the first element in the `args` must be the new context for the `fn`. 
That is why the initial value for the `applied` should be
```js
var applied = [null]
```

Sorry for not writing this explicitly the first time. 😞

Reply:
Ahhh, I'll fix it asap!